src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -887,6 +881,35 @@ LogicalResult ONNXSeluOp::inferShapes(
 
 LogicalResult ONNXSequenceInsertOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
+  onnxmlir::SeqType seqType =
+      input_sequence().getType().dyn_cast<mlir::onnxmlir::SeqType>();
+  ShapedType tensorType = tensor().getType().dyn_cast<ShapedType>();
+  if (seqType.getLength() == 0) {
+    getResult().setType(onnxmlir::SeqType::get(tensorType, 1));
+    return success();
+  }
+
+  auto seqShape = seqType.getElementType().cast<ShapedType>().getShape();
+  auto seqRank = seqType.getElementType().cast<ShapedType>().getRank();
+  if (seqRank == -1)
+    return success();
+
+  auto tensorShape = tensorType.getShape();
+  auto tensorRank = tensorType.getRank();
+  if (tensorRank != seqRank)
+    return success();
+  SmallVector<int64_t, 4> dims;
+  for (auto i = 0; i < tensorRank; i++) {
+    if (seqShape[i] != tensorShape[i]) {
```

Comment:
You could use the `test ? val1 : val2` to make the code more compact.
The approach is smart, I like it. It basically force possibly 2 different literal values to become "fake" runtime. Personally, I would like a comment explaining what you are doing here.